spike/CodeReaderDemo.py

```diff
@@ -1,100 +1,254 @@
-port = ''
-operation = ''
-number = ''
-speed = 0
-angle = 0
-
-def main():
-
-    global port
-    global operation
-    global number
-    
-    #Open racer file
-    racer = open("Racer.txt", "r")
-
-    #Define symbols
+class CodeParser():
+
+    #Parsing variables
+    PORT = ''
+    OPERATION = ''
+    NUMBER = ''
+    JUMP = ''
+    FUNC = ''
+    T1 = ''
+    T2 = ''
+    #FUNCTIONS = []
+
+    path = ''
+    SPEED = 0
+    ANGLE = 0
+
+    lineNum = 0
+
+    #Parsing libraries (or "keywords" if you wanna not sound like an arrogant prick)
     OPERATIONS = ['+', '-', '*', '/', '=']
     NUMBERS = ['0', '1', '2', '3', '4', '5', '6', '7', '8', '9']
     WHITESPACE = ' '
     PORTS = ["THROTTLE", "TURN"]
-    
-    #Break down symbols together
-    lexeme = ''
-    if racer.mode == "r":
-        code = racer.readlines()
-        for line in code:
-            port = ''
-            operation = ''
-            number = ''
-            lexeme = ''
-            for i, char in enumerate(line):
-                if char != WHITESPACE:
-                   lexeme += char
-                if (i + 1 < len(line)):
-                    if line[i + 1] == WHITESPACE or line[i + 1] == '\n':
-                        if lexeme in PORTS:
-                            port = lexeme
-                        if lexeme in OPERATIONS:
-                            operation = lexeme
-                        if bool([ele for ele in NUMBERS if(ele in lexeme)]):
-                            number = lexeme
-                        lexeme = ''
-                if (i == len(line) - 1):
-                    if lexeme in PORTS:
-                        port = lexeme
-                    if lexeme in OPERATIONS:
-                        operation = lexeme
-                    if bool([ele for ele in NUMBERS if(ele in lexeme)]):
-                        number = lexeme
-                    lexeme = ''
+
+
+    #Initialize the CodeParser
+    def __init__(self, path):
+        self.path = "Racer.txt"
+        self.SPEED = 0
+        self.ANGLE = 0
+
+        #Open racer file
+        self.racer = open(self.path, "r")
+
+        #Read lines in
+        if self.racer.mode == "r":
+            self.code = self.racer.readlines()
+            self.racer.close()
+
+        #Remove whitespace from file
+        line = 0
+        while line < len(self.code):
+            terms = self.code[line].split()
+            if len(terms) == 0:
+                self.code.pop(line)
+            line += 1
```

Comment:
This loop will skip two lines for every blank line. Calling `self.code.pop(line)` will decrease `len(self.code)` by one and then you add one to `line`. If there happens to be two consecutive blank lines, it will allow a blank line to make it through. Suppose `line = 2` and lines 2 and 3 are blank. Then `terms = None` for line 2. Line 2 is removed by `pop`. Line 3 becomes the new line 2. But we then increment `line` to 3, skipping over the new line 2. So `line += 1` should be in an `else` block.